eng/pipelines/templates/steps/pr-matrix-presteps.yml

```diff
@@ -38,6 +38,18 @@ steps:
       ArtifactPath: '$(Build.ArtifactStagingDirectory)/PackageInfoPublishing'
 
   - pwsh: |
+      # if Azure.Core is in the list of packages we want to always keep it around for validation regardless of whether or not it is a direct package
+      Get-ChildItem -Path "$(Build.ArtifactStagingDirectory)/PackageInfo" -Filter "Azure.Core.json" -Recurse |
+        ForEach-Object {
+          $json = Get-Content $_.FullName | ConvertFrom-Json
+
+          if ($json.IncludedForValidation -eq $true) {
+            Write-Host "Azure.Core is indirectly included, setting it to direct so that we build and analyze it."
+            $json.IncludedForValidation = $false
```

Comment:
Why does this have to be done here, vs. when we initially create these files? Seems like it's just always $false basically?